### PR TITLE
Move L7 CNP update test to new context section

### PR DESCRIPTION
**Pull request by joestringer**
Based upon ~~#5729~~.

Please review commit-by-commit. In particular, I don't understand why we were previously deleting the CNP before testing connectivity which seems racy. The first commit removes that logic, please review it.

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/6131)

### CODE REVIEW

**Comment by joestringer** [comment]
test-me-please

File: test/k8sT/Policies.go [hunk 1]
```diff
@@ -640,6 +614,112 @@ var _ = Describe("K8sPolicyTest", func() {
...
+				cnpUpdateDeny         = helpers.ManifestGet("cnp-update-deny-ingress.yaml")
+				cnpUpdateNoSpecs      = helpers.ManifestGet("cnp-update-no-specs.yaml")
+				cnpUpdateDenyLabelled = helpers.ManifestGet("cnp-update-deny-ingress-labelled.yaml")
+			)
+
+			validateL3L4 := func(allowApp3 bool) {
+				res := kubectl.ExecPodCmd(
+					helpers.DefaultNamespace, appPods[helpers.App2],
+					helpers.CurlFail(fmt.Sprintf("http://%s/public", clusterIP)))
+				ExpectWithOffset(1, res).Should(helpers.CMDSuccess(),
```
Comment (eloycoto) [inline test/k8sT/Policies.go]: fmt.Sprintf is not needed here

File: test/k8sT/Policies.go [hunk 2]
```diff
@@ -640,6 +614,112 @@ var _ = Describe("K8sPolicyTest", func() {
...
+					"%q cannot curl clusterIP %q",
+					appPods[helpers.App2], clusterIP)
+
+				res = kubectl.ExecPodCmd(
+					helpers.DefaultNamespace, appPods[helpers.App3],
+					helpers.CurlFail(fmt.Sprintf("http://%s/public", clusterIP)))
```
Comment (eloycoto) [inline test/k8sT/Policies.go]: fmt.Sprintf is not needed here

**Review by eloycoto — DISMISSED** [review]
Minor nitpick but all LGTM.

File: test/k8sT/Policies.go (hunk 1 above)
Comment (joestringer) [reply test/k8sT/Policies.go]: Where do I keep copy/pasting this from, I always make this mistake ;)

Comment (eloycoto) [reply test/k8sT/Policies.go]: Not all are updated TBH, only the new ones :-D

**Comment by joestringer** [comment]
test-me-please

EDIT: Failure:

```
23:23:45 • Failure [86.360 seconds]
23:23:45 K8sPolicyTest
23:23:45 /home/jenkins/workspace/Cilium-PR-Ginkgo-Tests-Validated/src/github.com/cilium/cilium/test/ginkgo-ext/scopes.go:351
23:23:45   Basic Test
23:23:45   /home/jenkins/workspace/Cilium-PR-Ginkgo-Tests-Validated/src/github.com/cilium/cilium/test/ginkgo-ext/scopes.go:351
23:23:45     Validate CNP update
23:23:45     /home/jenkins/workspace/Cilium-PR-Ginkgo-Tests-Validated/src/github.com/cilium/cilium/test/ginkgo-ext/scopes.go:351
23:23:45       Verifies that a CNP with L7 HTTP rules can be replaced with L7 Kafka rules [It]
23:23:45       /home/jenkins/workspace/Cilium-PR-Ginkgo-Tests-Validated/src/github.com/cilium/cilium/test/ginkgo-ext/scopes.go:383
23:23:45 
23:23:45       "app3-66c6cdf976-2qhp8" cannot curl to "10.102.17.56" public
23:23:45       Expected command: kubectl exec -n default app3-66c6cdf976-2qhp8 -- curl -s -D /dev/stderr --fail --connect-timeout 3 --max-time 8 http://10.102.17.56/public -w "time-> DNS: '%{time_namelookup}(%{remote_ip})', Connect: '%{time_connect}',Transfer '%{time_starttransfer}', total '%{time_total}'" 
23:23:45       To succeed, but it failed:
23:23:45       Exitcode: 28 
23:23:45       Stdout:
23:23:45        	 time-> DNS: '0.000015()', Connect: '0.000000',Transfer '0.000000', total '3.001087'
23:23:45       Stderr:
23:23:45        	 command terminated with exit code 28
23:23:45       	 
23:23:45       
23:23:45 
23:23:45       /home/jenkins/workspace/Cilium-PR-Ginkgo-Tests-Validated/src/github.com/cilium/cilium/test/k8sT/Policies.go:722
```

**Comment by joestringer** [comment]
test-me-please

**Review by ianvernon — APPROVED** [review]

**Pull request merged**